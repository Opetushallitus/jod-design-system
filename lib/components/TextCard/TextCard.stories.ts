import type { Meta, StoryObj } from '@storybook/react';

import { TextCard } from './TextCard';

const meta = {
  title: 'Primitives/Cards/TextCard',
  component: TextCard,
  tags: ['autodocs'],
} satisfies Meta<typeof TextCard>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/LP7Acqn5dNNHS0R1YUWM1n/cx_jod_sb?node-id=1-1059',
    },
    docs: {
      description: {
        story: 'This is a card component for displaying a text with a label.',
      },
    },
  },
  args: {
    text: 734,
    label: 'ammatteja',
  },
};

export const Longer: Story = {
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/LP7Acqn5dNNHS0R1YUWM1n/cx_jod_sb?node-id=1%3A1062',
    },
    docs: {
      description: {
        story: 'This is a card component for displaying a longer text with a label.',
      },
    },
  },
  args: {
    text: 2336,
    label: 'koulutuksia',
  },
};
