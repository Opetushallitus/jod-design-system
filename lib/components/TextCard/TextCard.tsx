export interface TextCardProps {
  /** Text shown on the card */
  text: string | number;
  /** Label shown on the card */
  label: string;
}

/**
 * This is a card component for displaying a text with a label.
 */
export const TextCard = ({ text, label }: TextCardProps) => {
  return (
    <div className="bg-jod-light flex w-min flex-col-reverse items-center rounded-[10px] px-[40px] py-[15px] font-bold">
      <span className="text-[24px] leading-[26.4px] text-[#4D5358]">{label}</span>
      <span className="text-jod-black text-[80px] leading-[88px]">{text}</span>
    </div>
  );
};
