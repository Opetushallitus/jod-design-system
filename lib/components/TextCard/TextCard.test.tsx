import { afterEach, describe, expect, it } from 'vitest';
import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/vitest';

import { TextCard } from './TextCard';

afterEach(() => {
  cleanup();
});

describe('TextCard', () => {
  const text = 'Lorem ipsum';
  const label = 'Dolor sit amet';

  it('renders the text and the label', () => {
    const { container } = render(<TextCard text={text} label={label} />);
    expect(screen.getByText(text)).toBeInTheDocument();
    expect(screen.getByText(label)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
