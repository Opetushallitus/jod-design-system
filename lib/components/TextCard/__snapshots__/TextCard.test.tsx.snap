// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextCard > renders the text and the label 1`] = `
<div
  class="flex w-min flex-col-reverse items-center rounded-[10px] bg-jod-light px-[40px] py-[15px] font-bold"
>
  <span
    class="text-[24px] leading-[26.4px] text-[#4D5358]"
  >
    Dolor sit amet
  </span>
  <span
    class="text-[80px] leading-[88px] text-jod-black"
  >
    Lorem ipsum
  </span>
</div>
`;
